python/lsst/ts/hvac/hvac_csc.py

```diff
@@ -78,32 +137,56 @@ async def connect(self):
         if self.connected:
             raise RuntimeError("Already connected")
 
+        # Initialize interal state track keeping
+        self._setup_hvac_state()
+
         if self.simulation_mode == 1:
-            # Connect to the Simulator Client.
+            # Use the Simulator Client.
             self.log.info("Connecting to SimClient.")
             self.mqtt_client = SimClient()
+            self.mqtt_client.start_publish_telemetry_every_second = (
+                self.start_telemetry_publishing
+            )
         else:
-            # Connect to the MQTT Client.
+            # Use the MQTT Client.
             self.log.info("Connecting to MqttClient.")
-            self.mqtt_client = MqttClient()
-
-        self.mqtt_client.connect()
-        self.telemetry_task = asyncio.create_task(self.get_telemetry())
+            host = self.config.host
+            port = self.config.port
+            self.mqtt_client = MqttClient(host, port)
+
+        await self.mqtt_client.connect()
+        if self.start_telemetry_publishing:
+            self.telemetry_task = asyncio.create_task(
+                self._publish_telemetry_regularly()
+            )
         self.log.info("Connected.")
 
     async def disconnect(self):
         """Disconnect the HVAQ client, if connected.
         """
-        self.log.info("Disconnecting")
-        self.telemetry_task.cancel()
         if self.connected:
-            self.mqtt_client.disconnect()
+            self.log.info("Disconnecting")
+            self.telemetry_task.cancel()
+            await self.mqtt_client.disconnect()
+
+    def _setup_hvac_state(self):
+        """Set up internal tracking of the MQTT state.
+        """
+        self.hvac_state = {}
+        mqtt_topics_and_items = xml.get_telemetry_mqtt_topics_and_items()
+        for mqtt_topic in mqtt_topics_and_items:
```

Comment:
I made the change to avoid the need to reference the collection twice, thanks.

And item is a string. I could add it to the InternalItemState class but I don't see the need for doing that since it is reflected by the dictionary key.